landing-page/src/components/landing-page-network.ts

```diff
@@ -0,0 +1,163 @@
+import "@material/mwc-linear-progress/mwc-linear-progress";
+import {
+  type CSSResultGroup,
+  LitElement,
+  type PropertyValues,
+  css,
+  html,
+  nothing,
+} from "lit";
+import { customElement, property, state } from "lit/decorators";
+import type {
+  LandingPageKeys,
+  LocalizeFunc,
+} from "../../../src/common/translations/localize";
+import "../../../src/components/ha-button";
+import "../../../src/components/ha-alert";
+import {
+  ALTERNATIVE_DNS_SERVERS,
+  getSupervisorNetworkInfo,
+  setSupervisorNetworkDns,
+} from "../data/supervisor";
+import { fireEvent } from "../../../src/common/dom/fire_event";
+
+const SCHEDULE_FETCH_NETWORK_INFO_SECONDS = 5;
+
+@customElement("landing-page-network")
+class LandingPageNetwork extends LitElement {
+  @property({ attribute: false })
+  public localize!: LocalizeFunc<LandingPageKeys>;
+
+  @state() private _networkIssue = false;
+
+  @state() private _getNetworkInfoError = false;
+
+  @state() private _dnsPrimaryInterface?: string;
+
+  protected render() {
+    if (!this._networkIssue && !this._getNetworkInfoError) {
+      return nothing;
+    }
+
+    if (this._getNetworkInfoError) {
+      return html`
+        <ha-alert alert-type="error">
+          <p>${this.localize("network_issue.error_get_network_info")}</p>
+        </ha-alert>
+      `;
+    }
+
+    return html`
+      <ha-alert
+        alert-type="warning"
+        .title=${this.localize("network_issue.title")}
+      >
+        <p>
+          ${this.localize("network_issue.description", {
+            dns: this._dnsPrimaryInterface || "?",
+          })}
+        </p>
+        <p>${this.localize("network_issue.resolve_different")}</p>
+        ${!this._dnsPrimaryInterface
+          ? html`
+              <p>
+                <b>${this.localize("network_issue.no_primary_interface")} </b>
+              </p>
+            `
+          : nothing}
+        <div class="actions">
+          ${ALTERNATIVE_DNS_SERVERS.map(
+            ({ translationKey }, key) =>
+              html`<ha-button
+                .index=${key}
+                .disabled=${!this._dnsPrimaryInterface}
+                @click=${this._setDns}
+                >${this.localize(translationKey)}</ha-button
+              >`
+          )}
+        </div>
+      </ha-alert>
+    `;
+  }
+
+  protected firstUpdated(_changedProperties: PropertyValues): void {
+    super.firstUpdated(_changedProperties);
+    this._fetchSupervisorInfo();
+  }
+
+  private _scheduleFetchSupervisorInfo() {
+    setTimeout(
+      () => this._fetchSupervisorInfo(),
+      SCHEDULE_FETCH_NETWORK_INFO_SECONDS * 1000
+    );
+  }
+
+  private async _fetchSupervisorInfo() {
+    try {
+      const response = await getSupervisorNetworkInfo();
+      if (!response.ok) {
+        throw new Error("Failed to fetch network info");
```

Comment:
We throw an error here, but we never catch or handle it?

Reply:
It is inside a try catch then we set error to true and show an error in the UI.

Reply:
I'm not awake yet I think... Keep your try blocks smaller btw, no need to wrap it all in a try.